src/Http/Request.php

```diff
@@ -88,7 +89,13 @@ public function getEvent(string $url, array $params = []): ResponseInterface
             'outlook.timezone="' . $requestOptions->getPreferredTimezone() . '"'
         ]));
 
-        return $this->connection->get($url, $requestOptions, ['skipQueryParams' => $params['skipQueryParams'] ?? true]);
+        $args = [
+            'skipQueryParams' => $params['skipQueryParams'] ?? true,
+            'RequestObj' => $this,
```

Comment:
May be you can use a setter inside connection to setup the request instance instead of passing it in args?